language/diem-tools/writeset-transaction-generator/src/release_flow/create.rs

```diff
@@ -35,29 +42,29 @@ pub fn create_release(
                 .iter()
                 .map(|(bytes, module)| (module.self_id(), bytes)),
         )?,
+        commit_hash: get_commit_hash()?,
+        diem_version: updated_diem_version,
+        release_name: release_name.to_string(),
     };
 
-    if first_release {
-        if load_artifact(&chain_id).is_ok() {
-            bail!("Previous release existed");
-        }
-        save_release_artifact(release_artifact.clone())?;
-    }
-    let artifact = load_artifact(&chain_id)?;
+    if !first_release {
+        let artifact = load_latest_artifact(&chain_id)?;
 
-    if artifact.chain_id != chain_id {
-        bail!("Artifact mismatch with on disk file");
-    }
-    if artifact.version > version {
-        bail!(
-            "Artifact version is ahead of the argument: old: {:?}, new: {:?}",
-            artifact.version,
-            version
-        );
+        if artifact.chain_id != chain_id {
+            bail!("Artifact mismatch with on disk file");
+        }
+        if artifact.version > version {
```

Comment:
Can we also check that `updated_diem_version > artifact.diem_version` (or perhaps better, `updated_diem_version == artifact.diem_version + 1`)?

Reply:
We can't check for `updated_diem_version == artifact.diem_version + 1` as the AOS can increment this version on their end as well. Will add the test.

Reply:
I see. Could we check against the current on-chain `diem_version` instead?